gfx_core.h

```diff
@@ -249,6 +249,16 @@ static inline GfxResult GFX_SET_ERROR_IMPL(GfxResult result, char const *file_na
     return result;
 }
 
+static inline void* gfxMalloc(size_t size)
+{
+#ifdef _MSC_VER
+    _Notnull_ /* Silence msvc warnings about not checking allocation failure */
+#endif
+    void *ret = malloc(size);
+    GFX_ASSERT(ret != nullptr); // out of memory
+    return ret;
+}
```

Comment:
To help cover allocation errors, I created this gfx malloc function. This allows for allocation failures to all be checked in a single location and to silence compiler warnings in one location.